Sources/DockProgress/DockProgress.swift

```diff
@@ -1,13 +1,34 @@
 import Cocoa
+import CoreVideo
 
 @MainActor
 public enum DockProgress {
-	private static var previousProgress: Double = 0
+    private static var t: Double = 0;
+    private static var displayLink: CVDisplayLink?
 	private static var progressObserver: NSKeyValueObservation?
 	private static var finishedObserver: NSKeyValueObservation?
 
 	private static let dockContentView = with(ContentView()) {
 		NSApp.dockTile.contentView = $0
+        
+        let displayLinkCallback: CVDisplayLinkOutputCallback = {(displayLink: CVDisplayLink, inNow: UnsafePointer<CVTimeStamp>, inOutputTime: UnsafePointer<CVTimeStamp>, flagsIn: CVOptionFlags, flagsOut: UnsafeMutablePointer<CVOptionFlags>, displayLinkContext: UnsafeMutableRawPointer?) -> CVReturn in
+            Task { await MainActor.run {
+                let speed = 1.0
+                t += speed * CVDisplayLinkGetActualOutputVideoRefreshPeriod(displayLink)
+                if (animatedProgress - progress).magnitude <= 0.01 {
+                    animatedProgress = progress
+                    t = 0
+                } else {
+                    animatedProgress = lerp(animatedProgress, progress, easeInOut(t));
+                }
+                updateDockIcon()
+            } }
```

Comment:
THis works fine. You simply forgot to remove a trailing brace last time.

```suggestion
            Task { @MainActor in
                let speed = 1.0
                t += speed * CVDisplayLinkGetActualOutputVideoRefreshPeriod(displayLink)
                if (animatedProgress - progress).magnitude <= 0.01 {
                    animatedProgress = progress
                    t = 0
                } else {
                    animatedProgress = lerp(animatedProgress, progress, easeInOut(t));
                }
                updateDockIcon()
            }
```

Reply:
Yes you're right! Sorry about that